File: Tubes/References/HPSource/Source16/PAS/leapyear.pas
{Leap year, for Turbo Pascal 3.x, http://rudih.info}

program BetterLeapyear(input, output);

var Year: integer;

LABEL
 loop1;

procedure RegularCalc(Year: integer);
{determines if year is a leap year.
 Only works for years not div by 100.}
  begin{RegularCalc}
   case (Year mod 4) of
    0: writeln('That year is a leap year.');
    3: writeln('The next year might be a leap year.');
    1, 2: writeln ('That year is not a leap year.')
   end {case}
  end;{regular calc}

procedure CenturyCalc(Year: integer);
{determines if year is a leap year.
 Only works for years div by 100.}
  begin{century calc}
   if (Year mod 400) = 0
     then writeln('That year is a leap year.')
     else writeln('Not even close to a leap year.')
  end; {century calc}

begin{program}
 loop1:;
 writeln('Enter a year. 0 exits.');
 readln(Year);
  if (Year mod 100) = 0
   then CenturyCalc(Year)
   else RegularCalc(Year);
 if (Year <> 0) then 
   goto loop1;
end. {program}

File: Tubes/References/HPSource/CPP_source_code/RH_Sep2014/leap_year.cpp
// Leap year, for C++ compilers, web: http://rudih.info
void regularcalc();
void centurycalc();
#include <iostream>

using namespace std;

int year, modyear;

main(){
 loop1:
 cout << "Enter a year. 0 exits." << endl;
 cin >> year;
 modyear=year%100;
 if (modyear == 0){
     centurycalc();
   }
   else {
     regularcalc();
   };
   if (year!=0) goto loop1;
};

void regularcalc(){
 modyear=year%4;
 if (modyear == 0) {
    cout << "That year is a leap year." << endl;
    }
    else {
      cout << "That year is not a leap year." << endl;
    };
};

void centurycalc(){
 modyear=year%400;
 if (modyear == 0) {
    cout << "That year is a leap year." << endl;
    }
    else {
      cout << "Not even close to a leap year." << endl;
    };
};

